bottlerocket/agents/src/bin/ec2-resource-agent/ec2_provider.rs

```diff
@@ -120,70 +120,73 @@ impl Create for Ec2Creator {
 
         // Determine the instance type to use. If provided use that one. Otherwise, for `x86_64` use `m5.large`
         // and for `aarch64` use `m6g.large`
-        let instance_type = if let Some(instance_type) = spec.configuration.instance_type {
-            instance_type
+        let instance_types = if !spec.configuration.instance_types.is_empty() {
+            spec.configuration.instance_types
         } else {
-            instance_type(&ec2_client, &spec.configuration.node_ami, &memo).await?
+            vec![instance_type(&ec2_client, &spec.configuration.node_ami, &memo).await?]
         };
-        info!("Using instance type '{}'", instance_type);
-
-        // Run the ec2 instances
-        let instance_count = spec
-            .configuration
-            .instance_count
-            .unwrap_or(DEFAULT_INSTANCE_COUNT);
-        info!("Creating {} instance(s)", instance_count);
```

Comment:
This should stay outside of the for loop.

Reply:
Line 131-134 definitely, but I thought we could use some kind of message mentioning the specific instance type we are currently trying per iteration of the loop.